test/LineIntersectionTest.cpp

```diff
@@ -0,0 +1,237 @@
+//
+// Created by Haico Dorenbos on 08-06-2020.
+//
+#include <gtest/gtest.h>
+#include <math.h>
+#include "roboteam_utils/HalfLine.h"
+#include "roboteam_utils/Line.h"
+#include "roboteam_utils/LineSegment.h"
+#include "roboteam_utils/Vector2.h"
+
+const double EPSILON = 1e-2; // Small difference that is still significant enough to really change a Vector2 position.
```

Comment:
You should note that vectors are similar if they have a Manhattan distance lower than 1e-5. This EPSILON has a quite high value, because you don't want to create similar vector positions.